packages/cli/src/commands/git/connect.ts

```diff
@@ -116,7 +117,7 @@ export default async function connect(
       confirm,
       org,
       project,
-      repoInfo: repoArg,
+      repoInfo: parsedUrlArg,
```

Comment:
I have reproduced the issue. After doing some testing, I believe this line (`repoInfo: parsedUrlArg,`) is the only line that needed to be changed. I suggest reverting all other changes.

Also, please add a test for this scenario to `packages/cli/test/unit/commands/git.test.ts`. The test needs to use a unlinked fixture, so there's a `unlinked` fixture you can use. The test should succeed if `vc git connect https://github.com/test_org/test_repo` correctly errors out failing to connect to `test_org/test_repo`.

Reply:
I have reverted the changes as requested but I still believe there is an issue with the types that could occur again if not fixed. And the double parsing is unnecessary.

--- 

For the test I tried my best but still can't run the codebase properly so it'll need to be double checked

Reply:
I agree the types can be improved, but I think that's beyond the scope of this issue. The double parsing isn't ideal, but it's ok in this moment.

As for getting the code base to work, be in the root of the repo, make sure you have Node 16 and pnpm 8, run `pnpm i && pnpm build`. That should complete without error. To run the test, `cd packages/cli`, then `pnpm test test/unit/commands/git.test.ts`.

Reply:
Unfortunately, I still can't build, I get this error when building `@vercel/edge`, which I assume comes from `esbuild`  😩 It might be because I have a fresh install and dependencies are not fixed in tsup but I don't know much about this library.


```
✘ [ERROR] Cannot start service: Host version "0.14.54" does not match binary version "0.14.47"
```